javatests/arcs/core/data/expression/EvaluatorParticleTest.kt

```diff
@@ -208,4 +208,13 @@ class EvaluatorParticleTest {
       bar.foos.sortedBy { it.value }.joinToString { it.value }
     }).containsExactly("(1-1)", "(2-1), (2-2)")
   }
+
+  @Test
+  fun numericWidening() = runHarnessTest(
+    NumericWideningTestHarness { EvaluatorParticle(NumericWideningRecipePlan.particles.first()) }
+  ) { harness ->
+    harness.input.dispatchStore(NumericWidening_Input(a = 3, b = 3.1415))
+    harness.start()
+    assertThat(harness.output.dispatchFetch()?.sum).isWithin(.0001).of(6.1415)
```

Comment:
If you're interested, Kotlin short/byte are also narrowed, so 

255b + 255b = 254 not 510

Reply:
Done!